DataAccess/Concrete/EntityFramework/Contexts/WebNetSampleDBContext.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using WebNetSample.Entity.Concrete;
+
+namespace DataAccess.Concrete.EntityFramework.Contexts
+{
+    public class WebNetSampleDBContext : DbContext
+    {
+        protected IConfiguration Configuration { get; set; }
+
+        public DbSet<Product> Products { get; set; }
+        public DbSet<Category> Categories { get; set; }
+        public DbSet<Supplier> Suppliers { get; set; }
+
+        public WebNetSampleDBContext()
+        {
+
+        }
+
+        public WebNetSampleDBContext(DbContextOptions dbContextOptions, IConfiguration configuration) : base(dbContextOptions)
+        {
+            Configuration = configuration;
+        }
+
+        //protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        //{
+        //    if (!optionsBuilder.IsConfigured)
+        //        base.OnConfiguring(optionsBuilder.UseSqlServer(Configuration.GetConnectionString("WebNetSampleConnectionString")));
+        //}
+
+        
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlServer(connectionString: @"Server=DESKTOP-5JKESUF\SQLEXPRESS;Database=WebNetSampleDB;Trusted_Connection=true");
```

Comment:
It gives an error when I remove that part. Because I guess [this](https://github.com/osmanaliyardim/WebNetSample/pull/1/files#diff-e07ce86344d18fc3235d2d2538dbf70ef36bf5c39e6903ab3b4bdba5497263f1R22) can not create connection and I don't know why right now. Do you have any suggestions?

Reply:
Hm, this is strange, it suppose to work correctly. Which error do you see? Have you tried to debug app start in _Program.cs_, maybe this row is not mapping during startup?

Reply:
Yeap, it seems like it jumps that part and It gives the error below:

> System.InvalidOperationException: 'No database provider has been configured for this DbContext. A provider can be configured by overriding the 'DbContext.OnConfiguring' method or by using 'AddDbContext' on the application service provider. If 'AddDbContext' is used, then also ensure that your DbContext type accepts a DbContextOptions<TContext> object in its constructor and passes it to the base constructor for DbContext.'

Reply:
Just a proposal: could you, please, parametrise your constructor `DbContextOptions` like:
```suggestion
           public WebNetSampleDBContext(DbContextOptions<WebNetSampleDBContext> dbContextOptions, IConfiguration configuration) : base(dbContextOptions)
```

Reply:
> public WebNetSampleDBContext(DbContextOptions<WebNetSampleDBContext> dbContextOptions, IConfiguration configuration) : base(dbContextOptions)

I already have that constructor just like you sent me

Reply:
Try to parametrise `DbContextOptions` with `WebNetSampleDBContext` to make sure it has the following view: `DbContextOptions<WebNetSampleDBContext> dbContextOptions`